src/virtualship/instruments/xbt.py

```diff
@@ -0,0 +1,125 @@
+"""XBT instrument."""
+
+from dataclasses import dataclass
+from datetime import timedelta
+from pathlib import Path
+
+import numpy as np
+from parcels import FieldSet, JITParticle, ParticleSet, Variable
+
+from ..spacetime import Spacetime
+
+
+@dataclass
+class XBT:
+    """Configuration for a single XBT."""
+
+    spacetime: Spacetime
+    min_depth: float
+    max_depth: float
+
+
+_XBTParticle = JITParticle.add_variables(
+    [
+        Variable("temperature", dtype=np.float32, initial=np.nan),
+        Variable("max_depth", dtype=np.float32),
+        Variable("min_depth", dtype=np.float32),
+        Variable("winch_speed", dtype=np.float32),
+    ]
+)
+
+
+def _sample_temperature(particle, fieldset, time):
+    particle.temperature = fieldset.T[time, particle.depth, particle.lat, particle.lon]
+
+
+def _xbt_cast(particle, fieldset, time):
+    # lowering
+    particle_ddepth = -particle.winch_speed * particle.dt
+    if particle.depth + particle_ddepth < particle.max_depth:
+        particle.raising = 1
+        particle_ddepth = -particle_ddepth
+
+
+def simulate_xbt(
+    fieldset: FieldSet,
+    out_path: str | Path,
+    xbts: list[XBT],
+    outputdt: timedelta,
+) -> None:
+    """
+    Use Parcels to simulate a set of XBTs in a fieldset.
+
+    :param fieldset: The fieldset to simulate the XBTs in.
+    :param out_path: The path to write the results to.
+    :param xbts: A list of XBTs to simulate.
+    :param outputdt: Interval which dictates the update frequency of file output during simulation
+    :raises ValueError: Whenever provided XBTs, fieldset, are not compatible with this function.
+    """
+    WINCH_SPEED = 1.0  # sink and rise speed in m/s
```

Comment:
Is this the typical fall speed of an XBT? Does it not depend on density? Can we do something more realistic here?

Reply:
Since XBTs measure temperature and time, what people used to do is to assume a free falling in the form of $z(t) = \alpha t + \beta t^2$.

$\alpha$ depend on the instrument specifications and $\beta$ is usually very small, taking into account the loss of weight. Following Kizu, Yoritaka and Hanawa (2005), $z(t) = 6.828 t - 0.00182 t^2$ for the T-5 (which is the one that goes deepest, as deep as 1830 m) and following Kizu, Sukigara and Hanawa (2011), $z(t) = 6.553 t - 0.00221 t^2$ for the most commonly used T-7 XBT (reaches 760 m).
Since $\beta$ is so small, maybe we can assume that they basically fall at a constant speed $\alpha$?  
What do you think?

Refs:
https://www.aoml.noaa.gov/phod/goos/meetings/2008/XBT/2005JO61115.pdf
https://os.copernicus.org/articles/7/231/2011/

Reply:
OK, thanks for clarifying. These are useful papers for background! 

I would stick as close to the 'real deal' as we can, so use a similar nomenclature as in the papers. 

And I agree that the quadratic term is small, but it's simple to calculate (we can keep track of time in the kernel) so why not add the term? Makes the VirtualShip even more realistic ;-)